components/HTTP/src/Websocket.h

```diff
@@ -0,0 +1,159 @@
+/**
+ * @file Websocket.h
+ * @author Locha Mesh Developers (contact@locha.io)
+ * @brief 
+ * @version 0.1
+ * @date 2020-03-25
+ * 
+ * @copyright Copyright (c) 2020 Locha Mesh Developers
+ * 
+ */
+
+
+#ifndef HTTP_WEBSOCKET_H
+#define HTTP_WEBSOCKET_H
+
+#include "Hex.h"
+#include <cJSON.h>
+#include <esp_https_server.h>
+
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+
+#include <stdint.h>
+#include <string>
+#include <vector>
+
+/**
+ * @brief customer structure
+ */
+struct client_data_t {
+    std::uint8_t shaUID[32]; // customer id
+    int timestamp;           // time the connection was made
+    bool is_alive;           // indicates if the connection is alive or not
+    int fd;                  // connection identifier
+};
+
+/**
+ * @brief message type
+ */
+enum class WsMsgType {
+    Handshake,
+    Message,
+    Status,
+    Action
+};
+
+
+struct async_resp_arg_t {
+    httpd_handle_t hd;
+    int fd;
+};
+
+/**
+ * @brief uid of incoming messages
+ */
+struct uid_message_t {
+    std::uint8_t from_uid[32]; // uid who's sending the message
+    std::uint8_t to_uid[32];  // id of who receives the message
+};
+
+
+class Websocket
+{
+public:
+    Websocket(Websocket const&) = delete; // Copy construct
+    Websocket(Websocket&&) = delete;      // Move construct
+
+    Websocket& operator=(Websocket const&) = delete; // Copy assign
+    Websocket& operator=(Websocket&&) = delete;      // Move assign
+
+    /**
+     * @brief Get the unique instance of the object
+     *
+     * @return Websocket&
+     */
+    static Websocket& getInstance()
+    {
+        static Websocket instance;
+        return instance;
+    }
+
+    /**
+     * @brief handles the data that comes from the websocket client
+     *
+     */
+    void onReceive(httpd_ws_frame_t ws_pkt, httpd_req_t* req);
+
+    /**
+     * @brief check active customers
+     *
+     */
+    void checkConnection();
+
+private:
+    /**
+   * @brief check active customers
+   *  
+   */
+    Websocket();
+
+
+    std::vector<client_data_t> m_client; // array of connected clients
+
+    httpd_req_t* req_handler; // http_esp_server connection handler
+
+    /**
+     * @brief  get message type
+     * @param  payload data sent by the client
+     * @return 
+     * 0 : when the message type is handshakey 
+     * 1 : when the message type is message
+     * 2 : when the message type is status 
+     * 4 : when the message type is action 
+     * -1: when the message type is invalid
```

Comment:
```suggestion
     * @return 0: when the message type is handshakey 
     * @return 1: when the message type is message
     * @return 2: when the message type is status 
     * @return 4: when the message type is action 
     * @return -1: when the message type is invalid
```

Reply:
I was reading how to do multiple returns and this is the correct way.